include/nlohmann/json.hpp

```diff
@@ -3717,6 +3719,10 @@ class basic_json
                  std::is_convertible<basic_json_t, ValueType>::value, int>::type = 0>
     ValueType value(const typename object_t::key_type& key, const ValueType& default_value) const
     {
+        // check defalut_value type
+        if(std::is_same<json::value_t, ValueType>::value) {
```

Comment:
I do not like runtime error. I think it makes more sense to disallow the type via SFINAE.

Reply:
nice suggestions😄